pilot/pkg/config/kube/gateway/testdata/deployment/istio-upgrade-to-1.24.yaml

```diff
@@ -0,0 +1,255 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  annotations:
+    gateway.istio.io/controller-version: "5"
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations: {}
+  labels:
+    gateway.istio.io/managed: istio.io-mesh-controller
+    gateway.networking.k8s.io/gateway-name: test-upgrade
+    topology.istio.io/network: network-1
+  name: test-upgrade
+  namespace: default
+  ownerReferences:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    name: test-upgrade
+    uid: ""
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations: {}
+  labels:
+    gateway.istio.io/managed: istio.io-mesh-controller
+    gateway.networking.k8s.io/gateway-name: test-upgrade
+    topology.istio.io/network: network-1
+  name: test-upgrade
+  namespace: default
+  ownerReferences:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    name: test-upgrade
+    uid: ""
+spec:
+  selector:
+    matchLabels:
+      gateway.networking.k8s.io/gateway-name: test-upgrade
+      istio.io/gateway-name: test-upgrade
```

Comment:
This label doesn't exist on the pods that are created so this seems broken.  I think we need to selectively add the `istio.io/gateway-name` to the spec.template.metadata.labels?

Reply:
right, i missed that